### PR TITLE
Efficient list remove

### PR DESCRIPTION
**Pull request by nielsenko**
Use indexOf in managed list remove

### CODE REVIEW

**Review by desistefanova — APPROVED** [review]
Shall we add something to changelog about this?

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 ### Enhancements
 * Added support for "frozen objects" - these are objects, queries, lists, or Realms that have been "frozen" at a specific version. All frozen objects can be accessed and queried as normal, but attempting to mutate them or add change listeners will throw an exception. `Realm`, `RealmObject`, `RealmList`, and `RealmResults` now have a method `freeze()` which returns an immutable version of the object, as well as an `isFrozen` property which can be used to check whether an object is frozen. ([#56](https://github.com/realm/realm-dart/issues/56))
 * You can now set a realm property of type `T` to any object `o` where `o is T`. Previously it was required that `o.runtimeType == T`. ([#904](https://github.com/realm/realm-dart/issues/904))
-* Performance of indexOf on realm lists has been improved. It now uses realm-core instead of the generic version from ListMixin. ([#911](https://github.com/realm/realm-dart/pull/911)). 
+* Performance of indexOf on realm lists has been improved. It now uses realm-core instead of the generic version from ListMixin. ([#911](https://github.com/realm/realm-dart/pull/911), [#915](https://github.com/realm/realm-dart/pull/915)).
```
Comment (nirinchev) [inline CHANGELOG.md:8]: Isn't this PR about `List.remove`?

File: lib/src/list.dart [hunk 2]
```diff
@@ -83,6 +83,17 @@ class ManagedRealmList<T extends Object?> with RealmEntity, ListMixin<T> impleme
...
   }
 
+  @override
+  bool remove(covariant T element) {
+    if (element is RealmObject && !element.isManaged) {
+      throw RealmStateError('Cannot call remove on a managed list with an element that is an unmanaged object');
+    }
+    final index = indexOf(element);
+    final found = index > 0;
+    if (found) removeAt(index);
```
Comment (nirinchev) [inline lib/src/list.dart:93]: Purely stylistic, but I hate those one-line ifs 😭

File: test/list_test.dart [hunk 3]
```diff
@@ -716,6 +716,20 @@ Future<void> main([List<String>? args]) async {
...
+      throwsA(isA<TypeError>()),
+    );
+
+    expect(
+      () => team.players.remove(Person('alice')),
+      throwsA(isA<RealmStateError>().having(
+        (e) => e.message,
+        'message',
+        'Cannot call remove on a managed list with an element that is an unmanaged object',
+      )),
```
Comment (nirinchev) [inline test/list_test.dart:731]: Isn't it simpler to use the generic throws:

```suggestion
      throws<RealmStateError>('Cannot call remove on a managed list with an element that is an unmanaged object'),
```

File: lib/src/list.dart (hunk 2 above)
Comment (nielsenko) [reply lib/src/list.dart:93]: dartfmt rules .. get over it

File: CHANGELOG.md (hunk 1 above)
Comment (nielsenko) [reply CHANGELOG.md:8]: 🤦

File: test/list_test.dart (hunk 3 above)
Comment (nielsenko) [reply test/list_test.dart:731]: Old habit

File: lib/src/list.dart (hunk 2 above)
Comment (nirinchev) [reply lib/src/list.dart:93]: You can always put `{` around to make it look nice 😜

**Review by nirinchev — APPROVED** [review]

**Review by blagoev — APPROVED** [review]

**Pull request merged**